### ISSUE TITLE
baseline (MNIST) type model 

### ISSUE THREAD

**Issue by kshannon** [kshannon/intracranial-hemorrhage-detection#6]
Goal is to get a baseline model trained and submitted this week.

**Comment by kshannon** [comment]
@tonyreina I think this would be good for chris to work on over the next few days. you guys agree?

**Comment by kshannon** [comment]
@tonyreina not sure if i messed up the batch size but this network is getting stuck at the end of the epoch, is there something dumb I did you can point out real quick? If not I will begin to look into this issue later today. Thanks! and good news is that the data seems to be of good quality in the train/validate.csv now since commit 530c9fe89b8f855527f8ca35fda5253ca3fa91c2

![stuck](https://user-images.githubusercontent.com/1323888/65904092-91003e80-e372-11e9-8bc9-3e238fa9aa25.png)

**Comment by tonyreina** [comment]
Is it stuck or just processing many, many validation images?

**Comment by kshannon** [comment]
never mind, you are correct @tonyreina , it was stuck on validation images for a while and yes the model was saved in models/ dir as a .pb file I will try to run incorporate it in the predict script if I have time this morning. If that works I will retrain the model with 10 epochs to get something real overnight. Exciting!

**Comment by kshannon** [comment]
@tonyreina @utcsox Right now the baseline model is working. However, it predicts the same array of probabilities for each image. I verified that each ```model.predict(img)``` call had unique data, and underwent the same preprocessing steps. I also checked the data flowing through and it seems like this might be a data/model issue rather than a pipeline issue. 

I will have to look at it with fresh eyes, but so far our end-to-end pipeline seems to be brittle, wonky, but working. We can do some slight work cleaning up code when we are together and maybe also looking into this issue. 

If I have time tomorrow I might try and set up a resnet50, also looking to incorporate the weighted log loss function. Once this submission script (slow because I am just doing the predict one img at a time) finishes I will submit it to kaggle.. see what happens haha.

**Comment by kshannon** [comment]
good news, the baseline model submission csv did upload w/o error and provide a score. bad news the model which predicts the same probabilities for each class between all images scored much worse than guessing 0.5. E.g. from the baseline submission csv
```
ID_acc6c836b_epidural | 0.07925984
ID_acc6c836b_intraparenchymal | 0.35502625
ID_acc6c836b_intraventricular | 0.2632258
ID_acc6c836b_subarachnoid | 0.35662878
ID_acc6c836b_subdural | 0.4692239
ID_acc6c836b_any | 0.876345
ID_dc846ac33_epidural | 0.07925984
ID_dc846ac33_intraparenchymal | 0.35502625
ID_dc846ac33_intraventricular | 0.2632258
ID_dc846ac33_subarachnoid | 0.35662878
ID_dc846ac33_subdural | 0.4692239
ID_dc846ac33_any | 0.876345
```

**Comment by kshannon** [comment]
Tony suggested model might be degenerating, switching to training on only class 1, script to create the csv is included in current master branch. tried upping learning rate and lowering it (0.01, 0.001, 0.0001) on only class 1 data, and with > 6 epochs, but the model is still holding its loss rate and not learning. Gonna try implementing a more complex model.

**Comment by kshannon** [comment]
@tonyreina  @utcsox  I Tried building off of Tony's baseline model and turned it into vgg16 (added cv2.resize() to dataloader) still not getting any sort of convergence.This model also predicts for every value:

[0. 1. 0. 0. 0. 1.] [00:06<1:02:24, 20.96it/s]

Training was done only on class one images:

![Screenshot from 2019-10-07 07-24-09](https://user-images.githubusercontent.com/1323888/66320358-8e9e6700-e8d3-11e9-9f1e-602764fc1fed.png)

**Comment by kshannon** [comment]
Implemented resnet50 for 15 epochs at 64x224x224x2 in 3bbba0f4462e71720d701629e76b2ee08888c683 so far the loss  has decreased a resonable amount 0.90 --> 0.65 halfway through epoch 1, with the acc slightly ticking up. I am training on the full dataset (not just class 1). Shall see how this looks tomorrow.

**Comment by kshannon** [comment]
fantastic!! model performed horridly, but the predictions for each img were different which shows that our end to end pipeline (while brittle) does work. Going to close this issue up and begin to work on feeding better quality data into our models (preprocessing, up/down sampling etc.)
